eZ/Publish/Core/MVC/Symfony/Matcher/ClassNameMatcherFactory.php

```diff
@@ -0,0 +1,159 @@
+     *
+     * @throws InvalidArgumentException
+     *
+     * @return \eZ\Publish\Core\MVC\Symfony\Matcher\MatcherInterface|\eZ\Publish\Core\MVC\Symfony\Matcher\ViewMatcherInterface
+     */
+    protected function getMatcher($matcherIdentifier)
+    {
+        // Not a FQ class name, so take the relative namespace.
+        if ($matcherIdentifier[0] !== '\\' && $this->matcherRelativeNamespace !== null) {
+            $matcherIdentifier = $this->matcherRelativeNamespace . "\\$matcherIdentifier";
+        }
+
+        // Retrieving the matcher instance from in-memory cache
+        if (isset($this->matchers[$matcherIdentifier])) {
+            return $this->matchers[$matcherIdentifier];
+        }
+
+        if (!class_exists($matcherIdentifier)) {
+            throw new InvalidArgumentException("Invalid matcher class '$matcherIdentifier'");
+        }
+        $this->matchers[$matcherIdentifier] = new $matcherIdentifier();
+
+        if ($this->matchers[$matcherIdentifier] instanceof RepositoryAwareInterface) {
+            $this->matchers[$matcherIdentifier]->setRepository($this->repository);
+        }
+
+        return $this->matchers[$matcherIdentifier];
+    }
+
+    /**
+     * Checks if $valueObject has a usable configuration for $viewType.
+     * If so, the configuration hash will be returned.
+     * $valueObject can be for example a Location or a Content object.
+     *
+     * @param \eZ\Publish\Core\MVC\Symfony\View\View $view
+     *
+     * @return array|null The matched configuration as a hash, containing template or controller to use, or null if not matched.
+     */
+    public function match(View $view)
+    {
+        $viewType = $view->getViewType();
+
+        if (!isset($this->matchConfig[$viewType])) {
+            return null;
+        }
+
+        if (!isset($this->alreadyMatched[$viewType])) {
+            $this->alreadyMatched[$viewType] = new SplObjectStorage();
+        }
+
+        // If we already matched, just returned the matched value.
+        if (isset($this->alreadyMatched[$viewType][$view])) {
+            return $this->alreadyMatched[$viewType][$view];
+        }
+
+        foreach ($this->matchConfig[$viewType] as $configHash) {
+            $hasMatched = true;
+            $matcher = null;
+            foreach ($configHash['match'] as $matcherIdentifier => $value) {
+                $matcher = $this->getMatcher($matcherIdentifier);
+                $matcher->setMatchingConfig($value);
+                if (!$matcher->match($view)) {
+                    $hasMatched = false;
+                }
+            }
+
+            if ($hasMatched) {
+                return $this->alreadyMatched[$viewType][$view] = $configHash + array('matcher' => $matcher);
+            }
+        }
+
+        return $this->alreadyMatched[$viewType][$view] = null;
+    }
+
+    /**
+     * @param array $matchConfig
+     *
+     * @return AbstractMatcherFactory
+     */
+    public function setMatchConfig($matchConfig)
```

Comment:
Hmmm, why would it be added to the interface ? Not all matcher factories have a match config, do they ?